vertical-pod-autoscaler/enhancements/8898-standardize-vpa-status-condition-handling/README.md

```diff
@@ -0,0 +1,172 @@
+- Add new conditions that indicate useful status to users
+
+### Non-Goals
+
+- Removing unused conditions from the API
+
+## Motivation
+
+The current VPA implementation handles conditions inconsistently - some conditions are deleted when they become "false" rather than having their status updated. This behavior deviates from [Kubernetes API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties), which recommend that conditions persist and toggle between `True` and `False` status values.
+
+This inconsistency causes several problems:
+
+1. **Monitoring and alerting**: Tools like Prometheus that watch for condition changes cannot reliably alert on VPA state because conditions appear and disappear rather than transitioning. Users cannot easily set up alerts like "alert if ConfigUnsupported has been True for more than 5 minutes" when the condition might not exist at all.
+
+2. **E2E testing**: Tests that verify VPA behavior must resort to waiting for arbitrary time periods rather than checking for specific condition states. With proper condition semantics, tests can wait for a condition like `ScalingRequired=False` to confirm that scaling has completed, providing faster and more reliable test results.
+
+3. **Observability**: Users and operators lack visibility into why a VPA is or isn't taking action. New conditions like `ScalingBlocked`, `ScalingRequired`, and `ScalingActionSucceeded` provide richer information about the VPA's decision-making process.
+
+## Proposal
+
+This proposal is essentially two parts: Improving existing conditions and adding new conditions.
+
+### Improve existing conditions
+
+Existing VPA conditions will be updated to persist with `status: False` instead of being deleted when their state becomes false. The following conditions are affected:
+
+- `ConfigDeprecated`
+- `ConfigUnsupported`
+- `NoPodsMatched`
+
+### Add new conditions
+
+In addition to changing existing conditions, this AEP also proposes to add new conditions. The purpose of this AEP is to retrofit conditions that should have always been there. The list is a work in progress and will be amended as new retrofitted conditions are required.
+
+Any new feature requiring new conditions will list those conditions in that feature's AEP.
+
+#### ScalingBlocked
+
+Type: ScalingBlocked
+
+True status reasons:
+
+- InsufficientReplicas
+- ScalingDisabled
+
+False status reasons:
+
+- SufficientReplicas
+
+#### ScalingRequired
+
+Type: ScalingRequired
+
+True status reasons:
+
+- PodResourcesDiverged (pod resources differ from recommendation)
+
+False status reasons:
+
+- NoEligiblePodsForScaling (no pods require scaling - resources match recommendation or no pods can be safely updated)
+
+#### ScalingActionSucceeded
+
+Type: ScalingActionSucceeded
```

Comment:
I'm wondering if we need this, as this seems more like an event than a condition to me.

For example, what if the last scaling action happened 5 days ago, but the user only looks at this condition, maybe ignoring that `ScalingBlocked` is actually true? I'm not entirely sure this reflects an actual current condition of the VPA object or not.